service/worker/esanalyzer/workflow.go

```diff
@@ -171,6 +216,123 @@ func (w *Workflow) emitWorkflowVersionMetrics(ctx context.Context) error {
 	return nil
 }
 
+func (w *Workflow) emitWorkflowVersionMetricsPinot(domainName string, logger *zap.Logger) error {
+	wfVersionPinotQuery, err := w.getWorkflowTypePinotQuery(domainName)
+	if err != nil {
+		logger.Error("Failed to get Pinot query to find workflow version Info",
+			zap.Error(err),
+			zap.String("DomainName", domainName),
+		)
+		return err
+	}
+	response, err := w.analyzer.pinotClient.SearchAggr(&pinot.SearchRequest{Query: wfVersionPinotQuery})
+	if err != nil {
+		logger.Error("Failed to query Pinot to find workflow type count Info",
+			zap.Error(err),
+			zap.String("VisibilityQuery", wfVersionPinotQuery),
+			zap.String("DomainName", domainName),
+		)
+		return err
+	}
+	foundAggregation := len(response) > 0
+
+	if !foundAggregation {
+		logger.Error("Pinot error: aggregation failed.",
+			zap.Error(err),
+			zap.String("Aggregation", fmt.Sprintf("%v", response)),
+			zap.String("DomainName", domainName),
+			zap.String("VisibilityQuery", wfVersionPinotQuery),
+		)
+		return err
+	}
+	var domainWorkflowVersionCount DomainWorkflowVersionCount
+	for _, row := range response {
+		workflowType := row[0].(string)
+		workflowCount, ok := row[1].(int)
+		if !ok {
+			logger.Error("Error parsing workflow count",
+				zap.Error(err),
+				zap.String("WorkflowType", workflowType),
+				zap.String("DomainName", domainName),
+			)
+			return fmt.Errorf("error parsing workflow count for workflow type %s", workflowType)
+		}
+		workflowVersions, err := w.queryWorkflowVersionsWithType(domainName, workflowType, logger)
+
+		if err != nil {
+			logger.Error("Error querying workflow versions",
+				zap.Error(err),
+				zap.String("WorkflowType", workflowType),
+				zap.String("DomainName", domainName),
+			)
+			return fmt.Errorf("error querying workflow versions for workflow type: %s: error: %s", workflowType, err.Error())
+		}
+
+		domainWorkflowVersionCount.WorkflowTypes = append(domainWorkflowVersionCount.WorkflowTypes, WorkflowTypeCount{
+			EsAggregateCount: EsAggregateCount{
+				AggregateKey:   workflowType,
+				AggregateCount: int64(workflowCount),
```

Comment:
workflowCount is from first call; this will be different from the summation of counts from subsequent calls by workflowtypes. But you could instead use the summation to be at least self consistent.

Reply:
Here's one sample result from ES:

`        {
          "key": "UpfrontChargeWorkflow::start",
          "doc_count": 182,
          "versions": {
            "doc_count_error_upper_bound": 0,
            "sum_other_doc_count": 0,
            "buckets": [
              {
                "key": "waitForPSPCallback-1",
                "doc_count": 149
              }
            ]
          }
        },
`

The count of workflow type is different from the summation of the counts of CadenceChangeVersions. I was thinking if this is designed on purpose.

Reply:
How about we group by WorkflowType and CadenceChangeVersion, so it can have the count per version and per type. I tried and it is working
```
SELECT JSON_EXTRACT_SCALAR(Attr, '$.CadenceChangeVersion', 'STRING_ARRAY') AS CadenceChangeVersion, COUNT(*) AS count, workflowtype
FROM rta.rta.cadence_visibility_production
WHERE IsDeleted = false
  AND CloseStatus = -1
  AND StartTime > 0
  AND JSON_EXTRACT_SCALAR(Attr, '$.CadenceChangeVersion', 'STRING_ARRAY') IS NOT NULL
GROUP BY JSON_EXTRACT_SCALAR(Attr, '$.CadenceChangeVersion', 'STRING_ARRAY'), workflowtype
ORDER BY count DESC
```

Reply:
That query means to count all the workflowTypes which has CadenceChangeVersion. This is different from the ES result.

Reply:
Discussed offline, group by version and type will filter the records without CadenceChangeVersion. Need to verify if we need to emit that count, if not we can go with this approach.